lib/note-list/no-notes.tsx

```diff
@@ -1,43 +1,62 @@
 import React from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 
+import NotesIcon from '../icons/notes';
+import TagIcon from '../icons/tag';
+import TrashIcon from '../icons/trash';
+
 import actions from '../state/actions';
 
 import * as S from '../state';
 
 const NoNotes = () => {
   const hasLoaded = useSelector((state: S.State) => state.ui.hasLoadedNotes);
   const searchQuery = useSelector((state: S.State) => state.ui.searchQuery);
+  const showTrash = useSelector((state: S.State) => state.ui.showTrash);
+  const openedTag = useSelector((state: S.State) => state.ui.openedTag);
   const dispatch = useDispatch();
 
-  const getMessage = () => {
-    return hasLoaded
-      ? searchQuery.length
-        ? 'No Results'
-        : 'No Notes'
-      : 'Loading Notes';
-  };
-
-  const getButton = () => {
-    return searchQuery.length ? (
+  const getSearchButton = () => {
+    return (
       <button
         onClick={() =>
           dispatch(actions.ui.createNote({ content: searchQuery }))
         }
       >
         Create a new note titled &quot;{searchQuery}&quot;
       </button>
-    ) : (
-      <button onClick={() => dispatch(actions.ui.createNote())}>
-        Create a new note
-      </button>
     );
   };
 
+  let message, icon, button;
+
+  switch (true) {
+    case searchQuery.length > 0:
+      message = 'No Results';
+      icon = null;
+      button = getSearchButton();
+      break;
+    case openedTag !== null:
+      message = `No notes tagged "${openedTag}"`;
+      icon = <TagIcon />;
+      button = '';
+      break;
+    case showTrash:
+      message = 'Your trash is empty';
+      icon = <TrashIcon />;
+      button = '';
+      break;
+    default:
+      message = 'Create your first note';
+      icon = <NotesIcon />;
+      button = '';
+  }
```

Comment:
this is clever, but we don't typically use the pattern of `switch (true)` and do'm reluctant to introduce it. it's definitively not what the purpose of the structure is for, plus this relies on mutating variables.

a simple approach to this scenario is to create a separate function and use `return` statements to abort or pick the values. a separate function also gives the benefit of making it easier to test the behavior

```ts
const placeholderInfo = ({searchQuery, openedTag, showTrash}) => {
	if (searchQuery.length > 0) {
		return {message: 'No Results', icon: null, button: getSearchButton()}
	}

	if (openedTag != null) {
		return {message: `No notes tagged "${openedTag}"`, icon: <TagIcon />, button: ''};
	}

	if (showTrash) {
		return {message: 'Your trash is empty', icon: <TrashIcon />, button: ''};
	}

	return {message: 'Create your first note', icon: <NotesIcon />, button: ''};
}

const {message, icon, button} = placeholderInfo({searchQuery, openedTag, showTrash});
```

adding types gives us additional safety, but I did not introduce those in this example

Reply:
This makes a lot of sense. I've updated to use this approach and added some types as well.